### PR TITLE
Per test timeout support

### PR DESCRIPTION
**Pull request by M4rtinK**
The first commit adds support for defining a per-test timeout override.

The second commit uses this new feature to set a 120 minute timeout for the Fedora live image build kickstart test.

A couple notes about thing's I'm not 100% sure about with this:
- what do you think about calling the bash function used for this `timeout()` ? any other suggestions ?
- what about input validation (so far we only split output from the bash functions in lists at most) ?
  - should we put the integer parsing into a try/except block and report issues or is this too niche for us to care ?
  - is the parsing to integers in the right place or would it be better to have it as a generic method in the `ShellOutput` class ?

### CODE REVIEW

**Comment by jkonecny12** [comment]
The solution is not bad  but I don't like the name `timeout`. It seems more like a callback than setter. So maybe something like `get_timeout`?

**Comment by jkonecny12** [comment]
Also seems that the travis found a bug in the code.

**Comment by rvykydal** [comment]
> The solution is not bad but I don't like the name `timeout`. It seems more like a callback than setter. So maybe something like `get_timeout`?

Yes I agree, moreover the function can conflict with timeout tool used in functions.sh:
https://github.com/rhinstaller/kickstart-tests/blob/b4445c885d6298b986b0f46c593900fbacf25ab4/functions.sh#L93
(eg if we add the function to the file, see my other comment)

**Comment by rvykydal** [comment]
> Also seems that the travis found a bug in the code.

I think it is caused that default definition of the function is missing in functions.sh
```
get_timeout() {
    echo ""
}
```

**Comment by rvykydal** [comment]
Other than the two comments I am fine with the PR as is.

**Comment by poncovka** [comment]
> Other than the two comments I am fine with the PR as is.

Same here.

**Comment by rvykydal** [comment]
@M4rtinK could you please look at the PR?

**Comment by M4rtinK** [comment]
> @M4rtinK could you please look at the PR?

I'll take a look during next week. :)

**Comment by M4rtinK** [comment]
Renamed the shell function to `get_timeout()` and added the stub to `functions.sh` - thanks for feedback! :)

**Review by jkonecny12 — APPROVED** [review]
Looks good to me now!

**Review by rvykydal — APPROVED** [review]
Looks good to me, thank you.

File: scripts/launcher/run_one_test.py [hunk 1]
```diff
@@ -135,6 +135,15 @@ def _create_virtual_conf(self, log_path) -> VirtualConfiguration:
...
         kernel_args = self._shell.kernel_args()
 
+        timeout_string = self._shell.get_timeout()
+        if timeout_string:
+            # looks like the test has a custom timeout set,
+            # try to parse it to an integer
+            timeout = int(timeout_string)
+        else:
+            # no custom timeout, fall back to the default 60 minute test timeout
+            timeout = 60
```
Comment (poncovka) [inline scripts/launcher/run_one_test.py:145]: The default value can be provided by `functions.sh`.

Comment (M4rtinK) [reply scripts/launcher/run_one_test.py]: Nice idea, let's do that. :)

**Review by poncovka — APPROVED** [review]

**Pull request merged**